### ISSUE TITLE
Speed up Patterns page

### ISSUE THREAD

**Issue by sashasushko** [moira-alert/moira#286]
We have hidden page with all patterns in Moira. Look at `/patterns`.

There is no pagination in API. And will not be. But sometimes there is hundreds of items in list. React can't fast render them. Page slow down.

Need some defer rendering.

### ToDo:
- [ ] add patterns fake data to fake api based on [json-server](https://www.npmjs.com/package/json-server). To simplify development
- [ ] add code for defer rendering

**Comment by lazycipher** [comment]
I would like to work on this.

**Comment by sashasushko** [comment]
Okay

**Comment by Vivekrajput20** [comment]
@lazycipher  are you still working on it? I would like to work on the issue if you aren't.

**Comment by lazycipher** [comment]
Hi @Vivekrajput20,

Yes, I'm working on it and I'll send PR soon.

**Comment by sashasushko** [comment]
@lazycipher 

Fake API based on [JSON server](https://www.npmjs.com/package/json-server). To add data you need edit `db.json`. Add in top level key `patterns`, value must be `list: []`. Like this:

```json
{
  "patterns": {
    "list": []
  }
}
```

Then add to list array few objects with data. Look at object structure in `Domain/Pattern.js`. You can do it manually or [write some code for data generation](https://www.npmjs.com/package/json-server#generate-random-data)

**Comment by lazycipher** [comment]
Hey @sashasushko ,
I'm following the same but I get an error when I use "patterns" but data loads on page when I use "pattern".

When I put `"patterns"` in db.json I get this -> [Imgur](https://i.imgur.com/pPXVvyd.png)
But when I put `"pattern"` and nativate to the patterns page, I get the data on page.

**Comment by sashasushko** [comment]
Yep. Use `pattern`. It's my mistake. I mixed up page URL and API URL

**Comment by Vivekrajput20** [comment]
How is it going @lazycipher ? Do you need any help?

**Comment by idoqo** [comment]
Hi @sashasushko I'd love to work on this and have created a PR for the fake data here: [#291]()

**Comment by EduardZaydler** [comment]
https://github.com/moira-alert/web2.0/pull/592
